### ISSUE TITLE
Is there a reason set_executable only sets S_IXUSR?

### ISSUE THREAD

**Issue by hartzell** [LLNL/spack#1483]
I was surprised to find that set_executable only made things executable for the owner.

As it is, I may end up doing this (from cuda.py) in my install step:

```
        chmod = which('chmod')
        chmod('+x', runfile)
```

This is more like what I thought set_executable would do:

```
diff --git a/lib/spack/llnl/util/filesystem.py b/lib/spack/llnl/util/filesystem.py
index 4cf9916..4d1a287 100644
--- a/lib/spack/llnl/util/filesystem.py
+++ b/lib/spack/llnl/util/filesystem.py
@@ -368,8 +368,11 @@ def traverse_tree(source_root, dest_root, rel_path='', **kwargs):


 def set_executable(path):
-    st = os.stat(path)
-    os.chmod(path, st.st_mode | stat.S_IEXEC)
+    mode = os.stat(path).st_mode
+    mode |= stat.S_IXUSR
+    mode |= stat.S_IXGRP
+    mode |= stat.S_IXOTH
+    os.chmod(path, mode)


 def remove_dead_links(root):
```

**Comment by hartzell** [comment]
@adamjstewart -- The *nextflow* package suffers from this issue, my users can't run `.../bin/nextflow` because its `0x0744`.

Should I just touch up the *nextflow* package or is this safe/worth addressing at the lower level?

**Comment by adamjstewart** [comment]
I don't see any reason why `set_executable` shouldn't set `a+x`. Can anyone else think of a reason?

P.S. I've used `chmod('+x',...)` before in the past, but mostly because I didn't know `set_executable` existed.

**Comment by hartzell** [comment]
Were you asking *me* about "anyone else think of a reason", or do we think the "others" are listening in?

**Comment by tgamblin** [comment]
I would be fine with making set_executable do `a+x`.  I think that makes more sense too.

**Comment by tgamblin** [comment]
One caveat: I actually think `set_executable` should only set the `x` bits for which the `r` bit is already set.  That is, it shouldn't grant `x` permissions to people who can't already read the file.

**Comment by hartzell** [comment]
Ooooo.  Bit twiddling.  Just like a job interview!

So the read-ability of the file is a side effect of the umask of the account doing the build, right.

Other Spack things I've installed are ending up (e.g. the git `.../bin/*`) 755, which might be from my umask (0022) or from git's install.

**Comment by adamjstewart** [comment]
Fixed by #3742.
